hazelcast/src/test/java/com/hazelcast/test/TestTaskExecutorUtil.java

```diff
@@ -19,68 +19,57 @@
 import com.hazelcast.spi.impl.PartitionSpecificRunnable;
 import com.hazelcast.spi.impl.operationservice.InternalOperationService;
-import com.hazelcast.util.ExceptionUtil;
 
 import java.util.concurrent.ArrayBlockingQueue;
 import java.util.concurrent.BlockingQueue;
 import java.util.concurrent.Callable;
 
 import static com.hazelcast.test.HazelcastTestSupport.getNodeEngineImpl;
+import static com.hazelcast.util.ExceptionUtil.sneakyThrow;
 import static java.util.concurrent.TimeUnit.SECONDS;
 
 /**
  * Convenience for executing {@link Callable} on a partition thread.
- * Unlike regular {@link PartitionSpecificRunnable} it's easy to return a value back to a caller.
- *
+ * <p>
+ * Unlike the regular {@link PartitionSpecificRunnable} it's easy to return a
+ * value back to a caller.
+ * <p>
  * This is intended to be used in tests only.
- *
  */
 public final class TestTaskExecutorUtil {
 
-    private static final Object NULL_VALUE = new Object();
     private static final int TIMEOUT_SECONDS = 120;
+    private static final Object NULL_VALUE = new Object();
 
     private TestTaskExecutorUtil() {
-
     }
 
     /**
-     * Executes a callable on a specific partition thread and return a result.
-     * This does NOT check if a given Hazelcast instance owns a specific partition.
+     * Executes a {@link Callable} on a specific partition thread and returns
+     * a result.
+     * <p>
+     * This does <b>not</b> check if a given Hazelcast instance owns a specific
+     * partition.
      *
-     * @param instance Hazelcast instance to be used for task executin
-     * @param task the task to be executed
+     * @param instance    Hazelcast instance to be used for task execution
+     * @param task        the task to be executed
      * @param partitionId selects partition thread
-     * @param <T> type of the result
+     * @param <T>         type of the result
      * @return result as returned by the callable
      */
     public static <T> T runOnPartitionThread(HazelcastInstance instance, final Callable<T> task, final int partitionId) {
         InternalOperationService operationService = getNodeEngineImpl(instance).getOperationService();
-        final BlockingQueue<Object> resultQueue = new ArrayBlockingQueue<Object>(1);
-        operationService.execute(new PartitionSpecificRunnable() {
-            @Override
-            public int getPartitionId() {
-                return partitionId;
-            }
-
-            @Override
-            public void run() {
-                try {
-                    Object result = wrapNullIfNeeded(task.call());
-                    resultQueue.add(result);
-                } catch (Throwable e) {
-                    resultQueue.add(e);
-                }
-            }
-        });
+        BlockingQueue<Object> resultQueue = new ArrayBlockingQueue<Object>(1);
+        operationService.execute(new PartitionSpecificRunnableWithResultQueue<T>(partitionId, task, resultQueue));
```

Comment:
I've pulled out the anonymous classes, since they are very painful to debug when dealing with multiple classloaders.